src/preact.d.ts

```diff
@@ -277,8 +284,8 @@ declare namespace JSX {
 		charSet?:string;
 		challenge?:string;
 		checked?:boolean;
-		class?:string;
-		className?:string;
+		class?:string | Object;
+		className?:string | { [key:string]: string | boolean | number };
```

Comment:
To be honest, I've never used `className: number`, but @developit says it can be number so it can be number :)